components/brave_wallet_ui/page/screens/fund-wallet/fund-wallet.tsx

```diff
@@ -0,0 +1,365 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { useHistory } from 'react-router'
+import {
+  useDispatch,
+  useSelector
+} from 'react-redux'
+
+// utils
+import { getLocale } from '../../../../common/locale'
+import { getTokensNetwork } from '../../../utils/network-utils'
+
+// types
+import {
+  BraveWallet,
+  UserAssetInfoType,
+  WalletAccountType,
+  WalletRoutes,
+  WalletState
+} from '../../../constants/types'
+
+// actions
+import { WalletActions } from '../../../common/actions'
+
+// options
+import { AllNetworksOption } from '../../../options/network-filter-options'
+import { SelectBuyOption } from '../../../components/buy-send-swap/select-buy-option/select-buy-option'
+
+// hooks
+import { usePrevNetwork } from '../../../common/hooks/previous-network'
+import { useMultiChainBuyAssets } from '../../../common/hooks/use-multi-chain-buy-assets'
+
+// style
+import { Column, Flex, LoadingIcon, Row, VerticalSpace } from '../../../components/shared/style'
+import { Description, MainWrapper, NextButtonRow, StyledWrapper, Title } from '../onboarding/onboarding.style'
+import {
+  ScrollContainer,
+  SearchWrapper,
+  SelectAssetWrapper
+} from './fund-wallet.style'
+
+// components
+import SearchBar from '../../../components/shared/search-bar'
+import SelectAccountItem from '../../../components/shared/select-account-item'
+import SelectAccount from '../../../components/shared/select-account'
+import { BuyAssetOptionItem } from '../../../components/shared/buy-option/buy-asset-option'
+import { CenteredPageLayout } from '../../../components/desktop/centered-page-layout/centered-page-layout'
+import { StepsNavigation } from '../../../components/desktop/steps-navigation/steps-navigation'
+import TokenLists from '../../../components/desktop/views/portfolio/components/token-lists'
+import { NavButton } from '../../../components/extension/buttons/nav-button/index'
+import CreateAccountTab from '../../../components/buy-send-swap/create-account'
+import SwapInputComponent from '../../../components/buy-send-swap/swap-input-component'
+import SelectHeader from '../../../components/buy-send-swap/select-header'
+
+export const FundWalletScreen = () => {
```

Comment:
Screen for selecting crypto assets to buy via on-ramp. (opens link to on-ramp partners)